regression/pages/ecommerce/redeem_coupon_page.py

```diff
@@ -0,0 +1,241 @@
+"""
+Redeem coupon page
+"""
+import re
+
+from bok_choy.page_object import PageObject
+
+from regression.pages.whitelabel.const import ECOMMERCE_URL_WITH_AUTH
+from regression.pages.common.utils import (
+    convert_date_format,
+    extract_numerical_value
+)
+
+
+def get_course_ids_from_link(link):
+    """
+    Get course Ids from link
+    Args:
+        link:
+    Returns:
+        Course Ids
+    """
+    course_id = re.search(r'(?<=asset-v1:)\w+\+\w+\+\w+', link)
+    course_id = course_id.group(0)
+    return 'course-v1:' + course_id
+
+
+class RedeemCouponPage(PageObject):
+    """
+    Redeem Coupon page
+    """
+
+    def __init__(self, browser, coupon_code):
+        """
+        Enrollment code has to be set by the test
+        """
+        super(RedeemCouponPage, self).__init__(browser)
+        self.coupon_code = coupon_code
+        self.course_tile = '.discount-multiple-courses>.box-shadow>div'
+
+    @property
+    def url(self):
+        """
+        Construct a URL to the page using the coupon code.
+        """
+        return (
+            ECOMMERCE_URL_WITH_AUTH +
+            'coupons/offer/?code=' +
+            self.coupon_code
+        )
+
+    def is_browser_on_page(self):
+        """
+        Is browser on the page?
+        Returns:
+            True if offer is present:
+        """
+        return self.q(css='#offer').present
+
+    def wait_for_course_tile(self):
+        """
+        Wait for course tiles to appear on page
+        :return:
+        """
+        self.wait_for_element_visibility(
+            self.course_tile, 'wait for course tile')
+
+    @property
+    def error_message(self):
+        """
+        Get the error message for invalid coupon
+        Returns:
+            error message:
+        """
+        return self.q(css='.depth.depth-2.message-error>h3').text[0]
+
+    @property
+    def course_info(self):
```

Comment:
I'm glad you used a dictionary here, but we have logic that can get course_name and course_org. You should reuse those methods, and then just have a separate method for getting the course_start_date.